receiver/mongodbatlasreceiver/receiver.go

```diff
@@ -77,47 +81,106 @@ func (s *mongodbatlasreceiver) shutdown(context.Context) error {
-		projects, err := s.client.Projects(ctx, org.ID)
+	var projectsWithConfig []ProjectWithConfig
+
+	if len(s.cfg.Projects) > 0 {
+		for _, projectCfg := range s.cfg.Projects {
+			project, err := s.client.GetProject(ctx, projectCfg.Name)
+			if err != nil {
+				s.log.Error("error retrieving project", zap.String("projectName", projectCfg.Name), zap.Error(err))
+				continue
+			}
+			projectsWithConfig = append(projectsWithConfig, ProjectWithConfig{Project: project, ProjectConfig: projectCfg})
+		}
+	} else {
+		orgs, err := s.client.Organizations(ctx)
 		if err != nil {
-			return fmt.Errorf("error retrieving projects: %w", err)
+			return fmt.Errorf("error retrieving organizations: %w", err)
 		}
-		for _, project := range projects {
-			nodeClusterMap, providerMap, err := s.getNodeClusterNameMap(ctx, project.ID)
+		for _, org := range orgs {
+			proj, err := s.client.Projects(ctx, org.ID)
 			if err != nil {
-				return fmt.Errorf("error collecting clusters from project %s: %w", project.ID, err)
+				s.log.Error("error retrieving projects", zap.String("orgID", org.ID), zap.Error(err))
+				continue
 			}
-
-			processes, err := s.client.Processes(ctx, project.ID)
-			if err != nil {
-				return fmt.Errorf("error retrieving MongoDB Atlas processes for project %s: %w", project.ID, err)
+			for _, project := range proj {
+				// Since there is no specific ProjectConfig for these projects, pass nil.
+				projectsWithConfig = append(projectsWithConfig, ProjectWithConfig{Project: project, ProjectConfig: nil})
 			}
-			for _, process := range processes {
-				clusterName := nodeClusterMap[process.UserAlias]
-				providerValues := providerMap[clusterName]
+		}
+	}
 
-				if err := s.extractProcessMetrics(ctx, time, org.Name, project, process, clusterName, providerValues); err != nil {
-					return fmt.Errorf("error when polling process metrics from MongoDB Atlas for process %s: %w", process.ID, err)
-				}
+	for _, projectWithConfig := range projectsWithConfig {
+		if err := s.processProject(ctx, time, projectWithConfig.Project, projectWithConfig.ProjectConfig); err != nil {
+			s.log.Error("error processing project", zap.String("projectID", projectWithConfig.Project.ID), zap.Error(err))
+		}
+	}
 
-				if err := s.extractProcessDatabaseMetrics(ctx, time, org.Name, project, process, clusterName, providerValues); err != nil {
-					return fmt.Errorf("error when polling process database metrics from MongoDB Atlas for process %s: %w", process.ID, err)
-				}
+	return nil
+}
 
-				if err := s.extractProcessDiskMetrics(ctx, time, org.Name, project, process, clusterName, providerValues); err != nil {
-					return fmt.Errorf("error when polling process disk metrics from MongoDB Atlas for process %s: %w", process.ID, err)
-				}
-			}
+func (s *mongodbatlasreceiver) processProject(ctx context.Context, time timeconstraints, project *mongodbatlas.Project, projectCfg *ProjectConfig) error {
+	org, err := s.client.GetOrganization(ctx, project.OrgID)
```

Comment:
We shouldn't have to do this again for projects which we obtained _from_ an organization i.e. the default situation.

Reply:
I suggest passing it in. We already have it when there are no projects lists. Otherwise, we have to look it up per project anyways.

Reply:
updated